### PR TITLE
fix: Call Carousel force update when children lenght changed

### LINKED ISSUE #16583: Carousel lost slick dot active status after request
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/embed/mm99klq82y)

### Steps to reproduce
The following steps simulate a normal http request
1. initialize Carousel with empty slides
2. send a http request
3. **father componet trigger rerender , with empty slides**
3. get the request back, and render with data.

```js
import React, { PureComponent } from "react";
import ReactDOM from "react-dom";
import "antd/dist/antd.css";
import "./index.css";
import { Carousel } from "antd";

class Test extends PureComponent {
  state = {
    projects: []
  };

  componentDidMount() {
    // 一秒后置空
    setTimeout(() => {
      this.setState({
        projects: []
      });
    }, 1000);

    // 两秒后设置数组后，默认选中丢失
    setTimeout(() => {
      this.setState({
        projects: [1, 2, 3, 4, 5]
      });
    }, 2000);
  }

  render() {
    const { projects } = this.state;
    return (
      <div>
        <Carousel>
          {projects.map(project => (
            <div>
              <h3>{project}</h3>
            </div>
          ))}
        </Carousel>
      </div>
    );
  }
}

export default Test;

ReactDOM.render(<Test />, document.getElementById("container"));
```

### What is expected?
Carousel renders the first slide and the first slick dot set active 
image

### What is actually happening?
Carousel renders the last slide and  Lost slick dot active status

image

| Environment | Info |
|---|---|
| antd | 3.18.1 |
| React | 16.9.0 |
| System | Mac |
| Browser | Chrome 74.0.3729.131 |

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

fix #16583

### 💡 Solution

Call Carousel force update when children length changed

### 📝 Changelog

Fix Carousel miss active status when Children changed

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by rdmclin2** [comment]
excellent!

File: components/carousel/index.tsx [hunk 1]
```diff
@@ -70,6 +71,12 @@ export default class Carousel extends React.Component<CarouselProps, {}> {
     this.innerSlider = this.slick && this.slick.innerSlider;
   }
 
+  componentDidUpdate(prevProps: CarouselProps) {
+    if (React.Children.count(this.props.children) !== React.Children.count(prevProps.children)) {
+      this.goTo(0, false);
```
Comment (rdmclin2) [inline components/carousel/index.tsx]: react-slick can set `initialSlide` prop，should `goTo(0)`  be set to ` goTo(initialSlide)`?  https://react-slick.neostack.com/docs/api#initialSlide

Comment (zombieJ) [reply components/carousel/index.tsx]: Greate~ Could you help to send a PR about this?

**Comment by yoyo837** [comment]
related: #16748

**Pull request merged**